File: java/hamming/src/main/java/Hamming.java
class Hamming {

    private String left; 
    private String right; 

    Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");  
        }

        left = leftStrand; 
        right = rightStrand; 
    }

    int getHammingDistance() {
        int distance = 0; 

        for (int i = 0; i < left.length(); i++) {
            if (left.charAt(i) != right.charAt(i)) 
                distance++; 
        }

        return distance; 
    }

}

File: javascript/hamming/hamming.js
export const compute = (left, right) => {
  if (!left.length && right) {
    throw new Error('left strand must not be empty');
  }

  if (!right.length && left) {
    throw new Error('right strand must not be empty');
  }

  if (left.length !== right.length) {
    throw new Error('left and right strands must be of equal length');
  }

  let result = 0;

  [...left].map((leftChar, index) => {
    if (leftChar !== right[index]) {
      result++;
    }
  });

  return result;
};
